optuna/storages/in_memory.py

```diff
@@ -236,6 +247,31 @@ def set_trial_value(self, trial_id, value):
 
             self.trials[trial_id] = self.trials[trial_id]._replace(value=value)
 
+    def update_cache(self, trial_id):
+        # type: (int) -> None
+
+        if self.trials[trial_id].state != structs.TrialState.COMPLETE:
+            return
+
+        if self.best_trial_id is None:
+            self.best_trial_id = trial_id
+            return
+        best_value = self.trials[self.best_trial_id].value
+        new_value = self.trials[trial_id].value
+        if best_value is None:
+            self.best_trial_id = trial_id
+            return
+        if new_value is None:
+            return
+        if (self.get_study_direction(IN_MEMORY_STORAGE_STUDY_ID) ==
+                structs.StudyDirection.MAXIMIZE):
+            if best_value < new_value:
+                self.best_trial_id = trial_id
+            return
+        if best_value > new_value:
+            self.best_trial_id = trial_id
```

Comment:
Please add test cases to confirm that `self.best_trial_id` is updated correctly whether the study direction is `MINIMIZE` or `MAXIMIZE`.

Reply:
The test cases will be added in #611.